taxi/taxibench_pandas_ibis.py

```diff
@@ -20,84 +20,74 @@
 )
 
 
-def validation_prereqs(omnisci_server_worker, data_files_names, files_limit, columns_names):
-    return omnisci_server_worker.import_data_by_pandas(
-        data_files_names=data_files_names, files_limit=files_limit, columns_names=columns_names,
-    )
-
+def run_queries(queries, parameters, etl_results, output_for_validation=None):
+    for query_name, query_func in queries.items():
+        query_result = query_func(**parameters[query_name])
+        etl_results[query_name] = (
+            query_result[0]
+            if isinstance(query_result, (tuple, list)) and len(query_result) == 2
+            else query_result
+        )
+        if output_for_validation is not None:
+            assert len(query_result) == 2
+            output_for_validation[query_name] = query_result[1]
 
-def run_queries(queries, parameters, etl_results):
-    for query_number, (query_name, query_func) in enumerate(queries.items()):
-        exec_time = query_func(**parameters)
-        etl_results[query_name] = exec_time
     return etl_results
 
 
 # Queries definitions
-def q1_ibis(table, df_pandas, queries_validation_results, queries_validation_flags, validation):
+def q1_ibis(table, input_for_validation):
     t_query = 0
     t0 = timer()
     q1_output_ibis = (
         table.groupby("cab_type").count().sort_by("cab_type")["cab_type", "count"].execute()
     )
     t_query += timer() - t0
 
-    if validation and not queries_validation_flags["q1"]:
+    if input_for_validation:
         print("Validating query 1 results ...")
 
-        queries_validation_flags["q1"] = True
-
-        q1_output_pd = df_pandas.groupby("cab_type")["cab_type"].count()
+        q1_output_pd = input_for_validation["Query1"]
 
         # Casting of Pandas q1 output to Pandas.DataFrame type, which is compartible with
         # Ibis q1 output
-        q1_output_pd_df = q1_output_pd.to_frame()
-        q1_output_pd_df.loc[:, "count"] = q1_output_pd_df.loc[:, "cab_type"]
-        q1_output_pd_df["cab_type"] = q1_output_pd_df.index
-        q1_output_pd_df.index = [i for i in range(len(q1_output_pd_df))]
-
-        # queries_validation_results["q1"] = compare_dataframes(
-        #     ibis_df=q1_output_pd_df,
-        #     pandas_df=q1_output_ibis,
-        #     pd=run_benchmark.__globals__["pd"],
-        # )
-        if queries_validation_results["q1"]:
-            print("q1 results are validated!")
+        q1_output_pd_data = {
+            q1_output_pd.name: list(q1_output_pd.index),
```

Comment:
```suggestion
            q1_output_pd.name: q1_output_pd.index.to_numpy(),
```
untested but should be faster and consume less memory